scripts/dev/tidy.php

```diff
@@ -0,0 +1,789 @@
+
+    Trim final newlines and allow up two or less final newlines:
+      <green>./tidy.php -N 2 --fix input-path</green>
+HELP;
+}
+
+/**
+ * Exclude these patterns from processing.
+ */
+function getBlacklist(): array
+{
+    return [
+        // Version control files
+        '/^\.git\/.+$/',
+
+        // Known binary files based on extension
+        '/^.+\.(a|bmp|bz2|data|exe|gd|gd2|gif|gz|ico|iff|jp2|jpc|jpeg|jpg)$/',
+        '/^.+\.(la|lo|mp3|mo|mov|o|obj|odt|otf)$/',
+        '/^.+\.(patch|pdf|phar|phar\.php|png|ppt|psd)$/',
+        '/^.+\.(so|svg|swf|tar|tar\.gz|tif|tiff|tgz|ttf)$/',
+        '/^.+\.(wbmp|webp|woff|woff2)$/',
+        '/^.+\.(zip)$/',
+
+        // Backup and hidden files
+        '/^.+\~$/',
+    ];
+}
+
+/**
+ * Filter and include these in processing after applying blacklist.
+ */
+function getWhitelist(): array
+{
+    return ['/^.+$/'];
+}
+
+/**
+ * When in PHP source code repository, also exclude these patterns.
+ */
+function getPhpBlacklist(): array
+{
+    return [
+        // Generated from other files
+        '/^Zend\/zend\_language\_scanner\.c$/',
+        '/^Zend\/zend\_ini\_scanner\.c$/',
+        '/^Zend\/zend\_vm\_execute\.h$/',
+        '/^Zend\/zend\_vm\_opcodes\.h$/',
+        '/^Zend\/zend\_vm\_opcodes\.c$/',
+        '/^sapi\/cli\/mime\_type\_map\.h$/',
+        '/^sapi\/phpdbg\/phpdbg\_lexer\.c$/',
+        '/^ext\/fileinfo\/data\_file\.c$/',
+        '/^ext\/json\/json\_scanner\.c$/',
+        '/^ext\/mbstring\/unicode\_data\.h$/',
+        '/^ext\/pdo\/pdo_sql_parser\.c$/',
+        '/^ext\/pdo_mysql\/php\_pdo\_mysql\_sqlstate.h$/',
+        '/^ext\/phar\/phar\_path\_check\.c$/',
+        '/^ext\/standard\/credits\_ext\.h$/',
+        '/^ext\/standard\/credits\_sapi\.h$/',
+        '/^ext\/standard\/html\_tables\.h$/',
+        '/^ext\/standard\/url\_scanner\_ex\.c$/',
+        '/^ext\/standard\/var\_unserializer\.c$/',
+        '/^win32\/cp_enc_map\.c$/',
+
+        // Bundled libraries maintained in the upstream repositories
+        '/^ext\/date\/lib/',
+        '/^ext\/fileinfo\/libmagic/',
+        '/^ext\/gd\/libgd/',
+        '/^ext\/mbstring\/oniguruma/',
+        '/^ext\/pcre\/pcre2lib/',
+        '/^ext\/sqlite3\/libsqlite/',
+
+        // Data
+        '/^ext\/standard\/html\_tables\/mappings/',
+
+        // Files maintained via Autotools
+        '/^TSRM\/m4\/ax\_func\_which\_gethostbyname\_r\.m4$/',
+        '/^build\/libtool\.m4$/',
+        '/^build\/ax\_check\_compile\_flag\.m4$/',
+        '/^build\/ax\_gcc\_func\_attribute\.m4$/',
+        '/^build\/shtool$/',
+        '/^ltmain\.sh$/',
+        '/^config\.guess$/',
+        '/^config\.sub$/',
+
+        // Visual Studio generated templates
+        '/^win32\/php7dllts\.rc$/',
+        '/^win32\/php7ts\.rc$/',
+        '/^win32\/php7ts\_cli\.rc$/',
+
+        // Known file types to exclude
+        '/^.+\.(phpt)$/',
+        '/^.*tests\/.*\.txt$/',
+        '/^.*tests\/.*\.res$/',
+        '/^.*tests\/.*\.xsl$/',
```

Comment:
I think the following 5 directories should exists in the blacklist (part of the Windows build system), from the root of php-src:

/Release -- NTS build x86
/Release_TS -- ZTS build x86
/Debug -- NTS debug build x86
/Debug_TS -- ZTS debug build x86
/x64 -- Contains any of the above 4 directories as subdirectories, if using an x64 compiler

Reply:
@KalleZ Added... The script otherwise now also ignores all items from the `.gitignore` file by default.

Reply:
Ah thats great \o/, thanks!